orttraining/orttraining/python/training/ortmodule/_custom_gradient_registry.py

```diff
@@ -109,6 +109,34 @@ def max_pool2d_gradient():
     ]
 
 
+def minmax_gradient():
+    # Gradient of torch.min(input) (and max)
+    return [
+        ('Equal', ['I(0)', 'O(0)'], ['Mask']),
+        ('Constant', [], ['Const_0'], {'value': {'value': 0., 'dtype': 'IElemType(0)', 'is_tensor': True}}),
+        ('Constant', [], ['Const_1'], {'value': {'value': 1., 'dtype': 'IElemType(0)', 'is_tensor': True}}),
+        ('Where', ['Mask', 'Const_1', 'Const_0'], ['MaskValue']),
+        ('ReduceSum', ['MaskValue'], ['MaskSum']),
```

Comment:
ONNX's ReduceSum's keepdim is true by default, if you don't set the keepdim attr to false here, the result will not be scalar, which might have perf impact to following calculation as it might need broadcasting offset calculation. Not sure if the binary elementwise already handle all-1 dims or not, better add the keepdim attr to false here.